doc/source/whatsnew/v0.22.0.txt

```diff
@@ -94,7 +94,7 @@ Bug Fixes
 Conversion
 ^^^^^^^^^^
 
--
+- Bug in :attr:`Timestamp.weekday_name` and :attr:`DatetimeIndex.weekday_name` not returning locale aware values (:issue:`12806`)
```

Comment:
I don't think we should this as a bug, it is rather an API change as we deliberately returned non-locale dependent values.

(and to be honest, I am also not sure that I really like locale-dependent return values)

Reply:
we could add an additonal accessor, e.g. ``weekday_name_for_locale``? (hate adding things, but I agree that this might not be only desirable in some cases).

Reply:
We could turn `weekday_name` and `month_name` into functions that can accept a `locale` argument and default it to  `'en_US.UTF8'`? I think that would be clean, but it would definitely be an API change.

Reply:
hmm, that's an idea. we could could make new ones ``weekday_names()`` (and then just deprecate ``weekday_name``